### ISSUE TITLE
[BUG] builtin.xmlPublicID condition not working properly

### ISSUE THREAD

**Issue by jmle** [konveyor/analyzer-lsp#799]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

latest

### Priority

Critical

### Current Behavior

It looks like the `builtin.xmlPublicID` condition is not working properly in rule `xml-glassfish-01000` for its test data.

### How Reproducible

Always (Default)

### Environment

```markdown
- **Analyzer Image SHA** 
- **JDTLS  base image SHA**
- **Provider Settings JSON**
```

**Comment by jmle** [comment]
@shawn-hurley I think some of those rules haven't been translated properly and therefore are empty, but the 6 of them should have the regex.

I guess we could create some `filePattern` rules that would do this on all XML files is that's what you mean, yes. We would have to remove this capability afterwards. I have created [this issue](https://github.com/konveyor/rulesets/issues/343) to track the translation.

**Comment by shawn-hurley** [comment]
I guess the question becomes, how does this capability differ from the xml one? IIUC this one will allow you to scope a search for something to the namespace. 

Say, asking for an element that may be in more than one `urn:jboss:domain` but only wanting to find the one in `urn:jboss:domain:datasources:ee:4.0` or something? 

Would that not still be a useful feature?
